### ISSUE TITLE
Support read-only dependencies

### ISSUE THREAD

**Issue by geekofalltrades** [gruntwork-io/terragrunt#1931]
I'm on a team that owns our full stack, and we're hoping to move full deployment of our stack to Terraform with the help of Terragrunt. We have an AWS account that we all share for development and staging. We have some Terraform that's run as a singleton for the account, which bootstraps the entire account for use by our team, and some that we run that's instanced once per developer, to set up per-developer instances of our environments for development. The latter Terraform relies on outputs from the former Terraform.

When developers are running their per-dev Terraform, they need to be careful to _not_ accidentally also run the singleton Terraform, as that Terraform isn't intended to change in the course of development and is depended on by the rest of the team.

It would be nice to have a way to natively fetch outputs in Terragrunt from the singletons without the risk that the singletons will actually be run.

I'm aware of `--terragrunt-strict-include`, but it comes with drawbacks for this use-case.

1. It's opt-in, rather than opt-out. All it takes is one dev to forget to use this flag when running Terragrunt, and they can potentially change singleton state relied on by the rest of the team. Now someone has to stop working and mount a cleanup operation to unblock the rest of the team. Even something like #1509 is opt-in and could be forgotten by a dev.
2. It's, frankly, tedious. The dev will be running `terragrunt [run-all] apply` repeatedly to iteratively test changes, and they will have to pass `--terragrunt-strict-include` every time, possibly multiple times per call depending on the folder structure.

What I envision for this is an option in the dependency block called `read_only`, or perhaps `skip_run` if you want it to match with `skip_outputs` more closely. This effectively translates the dependency block into a `terraform_remote_state` data source. Outputs can be read from it, but state-modifying commands are never run in it.

The way I'm planning to work around this right now is to write a data module that accumulates a bunch of `terraform_remote_state` data sources for all of the singleton Terragrunt modules. Devs will use dependencies on this module in place of dependencies on the regular singleton modules. It will provide a read-only interface into the singleton state.

**Comment by yorinasub17** [comment]
It doesn't quite make sense to me to label this on the `dependency` block, as the scope of the `dependency` block is `read-only` already. For example, running `terragrunt apply` on a single module does not propagate the `apply` to the dependencies.

This use case is supposed to be handled by the [skip](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#skip) attribute. Does it work if you set that to `true` on the modules you don't want to touch?

**Comment by geekofalltrades** [comment]
We have something like this folder structure:

```
prod
    global
    prod-1
    prod-2
    prod-3
stage
    global
    stage
dev
    dev
```

prod is in its own account. dev and stage share an account. The stuff in `stage/stage` and `dev/dev` are identical, and they **both** have dependencies on the stuff in `stage/global`. There is no `dev/global`, because the stuff in `stage/global` is configured once for the account, and dev and stage are sharing the account.

From `dev/dev`, I want to be able to have `dependency` blocks that reach back into `../../stage/global` to read outputs without the risk than anything in `stage/global` will actually run, because it shouldn't run in the course of dev. I can't put `skip` on the stuff in `stage/global`, because when we actually deploy stage, that stuff _does_ need to run.
